assets/js/components/KeyMetrics/MetricsSelectionPanel/index.js

```diff
@@ -35,39 +35,63 @@ import {
 	KEY_METRICS_SELECTION_FORM,
 	KEY_METRICS_SELECTION_PANEL_OPENED_KEY,
 } from '../constants';
+import useViewOnly from '../../../hooks/useViewOnly';
 import SideSheet from '../../SideSheet';
 import Header from './Header';
 import Footer from './Footer';
 import Metrics from './Metrics';
 const { useSelect, useDispatch } = Data;
 
 export default function MetricsSelectionPanel() {
+	const isViewOnly = useViewOnly();
+
 	const isOpen = useSelect( ( select ) =>
 		select( CORE_UI ).getValue( KEY_METRICS_SELECTION_PANEL_OPENED_KEY )
 	);
-	const keyMetrics = useSelect( ( select ) => {
+	const savedMetrics = useSelect( ( select ) => {
 		const metrics = select( CORE_USER ).getKeyMetrics();
 
 		if ( ! Array.isArray( metrics ) ) {
 			return [];
 		}
 
-		const { isModuleConnected } = select( CORE_MODULES );
+		const { getModule } = select( CORE_MODULES );
 		const { getWidget } = select( CORE_WIDGETS );
+		const { canViewSharedModule } = select( CORE_USER );
 
-		// Before setting metrics as selected, verify that they are available, i.e.
-		// the modules that they depend on are connected. This helps prevent metrics
-		// with disconnected dependencies appear as selected in the selection panel.
-		return metrics.filter( ( slug ) => {
-			const widget = getWidget( slug );
+		// Verify that the metrics are available, i.e. the modules that they depend
+		// on are connected and a view-only user has access to it. This helps prevent
+		// metrics with disconnected dependencies appear in the selection panel.
+		return metrics.filter( ( metric ) => {
+			const widget = getWidget( metric );
 
 			if ( ! widget ) {
 				return false;
 			}
 
-			return widget.modules.every( ( module ) =>
-				isModuleConnected( module )
-			);
+			const isMetricAvailable = widget.modules.every( ( slug ) => {
```

Comment:
We should be careful when using `slug` as a name in a widget context because it could refer to many different things (context/area/widget/module/...) which is why we tend to prefix slugs with the kind of slug it is to avoid ambiguity – particularly in the widgets API because there are so many possibilities 😄

Reply:
Used `widgetSlug` to reference this in the selector, thank you!